python/src/llvm.cc

```diff
@@ -239,6 +239,7 @@ void init_triton_llvm(py::module &&m) {
         }
         std::string obj = translateLLVMIRToASM(
             *module, triple, proc, features, flags, enable_fp_fusion, isObject);
+	py::gil_scoped_acquire no_threads;
```

Comment:
Minor issue: this results in an unnecessary sequence of `unlock -> lock -> unlock -> lock` because `gil_scoped_acquire` releases the GIL in its destructor. It would be better to move the `gil_scoped_release` into a more narrow scope, e.g.
```cpp
{
  py::gil_scoped_release allow_threads;
  ...
}
return isObject ? py::bytes(obj) : py::str(obj);
```